crates/p2p-network/examples/p2p.rs

```diff
@@ -0,0 +1,54 @@
+//! To be run manually to test auto-reconnect.
+
+use bytes::Bytes;
+use futures_util::StreamExt;
+use gw_config::P2PNetworkConfig;
+use gw_p2p_network::*;
+use tentacle::{builder::MetaBuilder, multiaddr::Protocol, service::ProtocolMeta, ProtocolId};
+
+const PROTOCOL_ZERO: ProtocolId = ProtocolId::new(0);
+
+fn protocol() -> ProtocolMeta {
+    MetaBuilder::new()
+        .id(PROTOCOL_ZERO)
+        .protocol_spawn(FnSpawn(|ctx, control, mut read| {
```

Comment:
The protocol name must be specified, it is the place where compatibility is guaranteed, the flag of the handshake

Reply:
OK. This is an example for testing. I will set names for the real protocols.

Reply:
> compatibility is guaranteed

So how does this really work? How does the tentacle API support it? Say I want to upgrade a protocol but maintain backward compatibility with the origin protocol, is that what this is about?

Reply:
If the protocol name is the same, it will open with the latest common version.
If the corresponding protocol name is not found, then it will not be opened.

Current ckb supports 2 versions of protocol open at the same time:
https://github.com/nervosnetwork/ckb/blob/develop/network/src/protocols/support_protocols.rs#L98

When you open it, you can know the open version: https://github.com/nervosnetwork/ckb/blob/develop/network/src/protocols/discovery/mod.rs#L74

Reply:
There are two ways to maintain backward compatibility:
1. distinguish the behavior of the protocol with the version number and, after a period of parallelism, remove the support of the old version
2. reopen a new protocol and keep the old protocol open, and after a parallel period, turn off support for the old protocol

Reply:
OK. Got it.